### PR TITLE
Add all of the no-op segmentation mask augmentations

### PR DESCRIPTION
**Pull request by ianstenbit**
This adds "segmentation map support" to all the layers that shouldn't change a segmentation map (mostly because they're just color adjustments)

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

For features like this, we want to make sure that we don't forget to implement support for the feature in new layers that we ship. This requires having a unit test that automatically checks the presence of the feature (similar to `get_config()` checks in the standard layer test).

**Comment by qlzh727** [comment]
Agreed. We probably want to have a test like https://github.com/keras-team/keras-cv/blob/master/keras_cv/layers/preprocessing/with_labels_test.py for segmentation_map. Basically a layer_test for all the common functionality of image KPLs.

**Review by qlzh727 — CHANGES_REQUESTED** [review]
Please add a test case for common no-op layers.

File: keras_cv/layers/preprocessing/with_segmentation_masks_test.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+class WithSegmentationMasksTest(tf.test.TestCase, parameterized.TestCase):
+    @parameterized.named_parameters(*TEST_CONFIGURATIONS)
+    def test_can_run_with_segmentation_masks(self, layer_cls, init_args):
+        layer = layer_cls(**init_args)
+
+        img = tf.random.uniform(
+            shape=(3, 512, 512, 3), minval=0, maxval=1, dtype=tf.float32
+        )
+        segmentation_masks = tf.random.uniform(
+            shape=(3, 512, 512, 3), minval=0, maxval=1, dtype=tf.int32
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/with_segmentation_masks_test.py:82]: The shape of the mask should probably be 3, 512, 512, 1

**Comment by ianstenbit** [comment]
I've added a with_segmentation_mask test, but I'm not sure that this fully solves the problem -- we still need to manually make sure these tests include all of the relevant layers (which the with_labels test currently did not).

These tests also don't assert that the segmentation mask and/or label are actually passed through in the output, so I've updated the with_labels test to do that and done the same in with with_segmentation_mask test.

File: keras_cv/layers/preprocessing/with_segmentation_masks_test.py (hunk 1 above)
Comment (qlzh727) [inline keras_cv/layers/preprocessing/with_segmentation_masks_test.py:82]: The value range should be 0 ~ classes, and 1 seems to be a bad choice here.

File: keras_cv/layers/preprocessing/with_segmentation_masks_test.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        )
+
+        inputs = {"images": img, "segmentation_masks": segmentation_masks}
+        _ = layer(inputs)
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/with_segmentation_masks_test.py]: We should validate that the output mask is not changed (no op).

File: keras_cv/layers/preprocessing/with_segmentation_masks_test.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    # This has to be a separate test case to exclude CutMix and MixUp
+    # (which are not yet supported for segmentation mask augmentation)
+    @parameterized.named_parameters(*TEST_CONFIGURATIONS)
+    def test_can_run_with_segmentation_mask_single_image(self, layer_cls, init_args):
+        layer = layer_cls(**init_args)
+        img = tf.random.uniform(
+            shape=(512, 512, 3), minval=0, maxval=1, dtype=tf.float32
+        )
+        segmentation_mask = tf.random.uniform(
+            shape=(512, 512, 3), minval=0, maxval=1, dtype=tf.int32
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/with_segmentation_masks_test.py:99]: same here.

**Review by qlzh727 — CHANGES_REQUESTED** [review]

**Comment by qlzh727** [comment]
I think we probably  

> I've added a with_segmentation_mask test, but I'm not sure that this fully solves the problem -- we still need to manually make sure these tests include all of the relevant layers (which the with_labels test currently did not).
> 
> These tests also don't assert that the segmentation mask and/or label are actually passed through in the output, so I've updated the with_labels test to do that and done the same in with with_segmentation_mask test.

File: keras_cv/layers/preprocessing/with_segmentation_masks_test.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/layers/preprocessing/with_segmentation_masks_test.py:82]: Done

Comment (ianstenbit) [reply keras_cv/layers/preprocessing/with_segmentation_masks_test.py:82]: Done

File: keras_cv/layers/preprocessing/with_segmentation_masks_test.py (hunk 2 above)
Comment (ianstenbit) [reply keras_cv/layers/preprocessing/with_segmentation_masks_test.py]: Done, although this won't always be true (once we add non-noop KPLs for seg masks). I've left a comment about this, and we can add a parameter for this to the tests in the future.

File: keras_cv/layers/preprocessing/base_image_augmentation_layer.py [hunk 4]
```diff
@@ -28,7 +28,7 @@
 BOUNDING_BOXES = "bounding_boxes"
 KEYPOINTS = "keypoints"
 RAGGED_BOUNDING_BOXES = "ragged_bounding_boxes"
-SEGMENTATION_MASK = "segmentation_mask"
+SEGMENTATION_MASK = "segmentation_masks"
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/base_image_augmentation_layer.py:31]: SEGMENTATION_MASK[S]?

Comment (ianstenbit) [reply keras_cv/layers/preprocessing/base_image_augmentation_layer.py]: Done. Made it plural for consistency with the other labels

**Review by qlzh727 — APPROVED** [review]

**Review by LukeWood — APPROVED** [review]
Looks good to me, thanks Ian!

**Pull request merged**